utopia-remix/app/routes/projects.tsx

```diff
@@ -47,119 +44,129 @@ const ProjectsPage = React.memo(() => {
-        break
-      default:
-        assertNever(selectedCategory)
+  const fetcher = useFetcher()
+
+  const [projects, setProjects] = React.useState<ProjectWithoutContent[]>(data.projects)
+  const [searchQuery, setSearchQuery] = useState<string>('')
+  const [isDarkMode, setIsDarkMode] = useState<boolean>(false)
+
+  const filteredProjects = React.useMemo(() => {
+    const sanitizedQuery = searchQuery.trim().toLowerCase()
+    if (sanitizedQuery.length === 0) {
+      return projects
     }
-  }, [selectedCategory, data.projects, data.deletedProjects])
+    return projects.filter((project) => project.title.toLowerCase().includes(sanitizedQuery))
+  }, [projects, searchQuery])
 
-  React.useEffect(() => {
-    updateProjects()
-  }, [updateProjects])
+  const selectedProjectId = useStore((store) => store.selectedProjectId)
+  const setSelectedProjectId = useStore((store) => store.setSelectedProjectId)
+  const selectedCategory = useStore((store) => store.selectedCategory)
+  const setCategory = useStore((store) => store.setSelectedCategory)
 
-  const filterProjects = React.useCallback(() => {
-    if (searchValue === '') {
-      setFilteredProjects(projects)
-      setSearchQuery('')
-    } else {
-      const filteredProjects = projects.filter((project) =>
-        project.title.toLowerCase().includes(searchValue.toLowerCase()),
-      )
-      setFilteredProjects(filteredProjects)
-      setSearchQuery(searchValue)
+  const updateProjects = React.useCallback(
+    (category: Category) => {
+      switch (category) {
+        case 'allProjects':
+          setProjects(data.projects)
+          break
+        case 'trash':
+          setProjects(data.deletedProjects)
+          break
+        default:
+          assertNever(category)
+      }
+    },
+    [data.projects, data.deletedProjects],
+  )
+
+  const handleEmptyTrash = React.useCallback(() => {
+    const ok = window.confirm(
+      'Are you sure? ALL projects in the trash will be deleted permanently.',
+    )
+    if (ok) {
+      fetcher.submit({}, { method: 'POST', action: `/internal/projects/destroy` })
     }
-  }, [projects])
+  }, [fetcher])
+
+  const handleProjectSelect = React.useCallback(
+    (project: ProjectWithoutContent) => {
+      if (project.deleted) {
+        return
+      }
+      setSelectedProjectId(project.proj_id === selectedProjectId ? null : project.proj_id)
+    },
+    [selectedProjectId, setSelectedProjectId],
+  )
+
+  const udpateCategory = React.useCallback(
+    (category: Category) => {
+      setCategory(category)
+      setSearchQuery('')
+      setSelectedProjectId(null)
+      updateProjects(category)
+    },
+    [selectedCategory, setSelectedProjectId, updateProjects],
+  )
+
+  const handleCategoryClick = React.useCallback(
+    (category: React.SetStateAction<string>) => {
+      udpateCategory(category as Category)
+    },
+    [udpateCategory],
+  )
 
+  // when the media query changes, update the theme
```

Comment:
this should be a nicely named hook, so we can reuse across pages and spare the comment :)